src/NzbDrone.Core/Update/InstallUpdateService.cs

```diff
@@ -164,13 +219,20 @@ public void Execute(ApplicationUpdateCommand message)
 
             if (latestAvailable != null)
             {
-                InstallUpdate(latestAvailable);
+                ExecuteInstallUpdate(message, latestAvailable);
             }
         }
 
         public void Execute(InstallUpdateCommand message)
         {
-            InstallUpdate(message.UpdatePackage);
+            var latestAvailable = _checkUpdateService.AvailableUpdate();
+
+            if (latestAvailable == null || latestAvailable.Hash != message.UpdatePackage.Hash)
+            {
+                throw new ApplicationException("Unknown or invalid update specified");
```

Comment:
Previously we accepted a POST of whatever was submitted, which could have resulted in a malicious package being installed.

If the update isn't the latest the messaging should say that, entirely possible that it happens due to the user sitting on the update page for a while and a new update is available (develop in particular).